src/Molecules/Input/FormattedNumber/FormattedInput/utils.ts

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import { IntlShape } from 'react-intl';
+import { InputValue } from './FormattedInput.types';
+
+const escapeStringRegexp = (s: string) =>
+  s.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&').replace(/-/g, '\\x2d');
+
+export const getDecimalSign = (intl: IntlShape): string => intl.formatNumberToParts(0.1)[1].value;
+
+/** Retrieves locale specific thousand separator sign */
+export const getThousandsSeparator = (intl: IntlShape): string =>
+  intl.formatNumberToParts(1000)[1].value;
+
+/**
+ * Retrieves the locale specific minus sign.
+ * Certain locales, such as sv-SE, may have a different minus signs.
+ */
+export const getMinusSign = (intl: IntlShape): string => intl.formatNumberToParts(-1)[0].value;
+
+export const formatNumber = (value: number, intl: IntlShape, minimumFractionDigits = 0) =>
+  intl.formatNumber(value, { minimumFractionDigits, maximumFractionDigits: 20 });
+
+const getSanitizedValue = (inputValue: string, replaceValue: string, intl: IntlShape) => {
+  const decimalSign = getDecimalSign(intl);
+  const minusSign = getMinusSign(intl);
+
+  const acceptDotAsDecimalSign = getThousandsSeparator(intl) !== '.' ? '.' : '';
+  const nonNumericRegExp = new RegExp(
+    `[^${minusSign}${decimalSign}\\d-${acceptDotAsDecimalSign}]`,
+    'g',
+  );
+  const replaceDecimalRegExp = new RegExp(escapeStringRegexp(decimalSign), 'g');
+  const replaceMinusSignRegExp = new RegExp(escapeStringRegexp(minusSign), 'g');
+
+  const numericValue = inputValue.replace(nonNumericRegExp, replaceValue);
+
+  const delocalizedValue = numericValue
+    .replace(replaceDecimalRegExp, '.') // replace localized decimalSign
+    .replace(replaceMinusSignRegExp, '-'); // replace localized minusSign
+
+  const sanitizedValue = delocalizedValue
+    .replace(/(?<=^|-)\./, '0.') // prepend 0 when integer part is missing for decimal
```

Comment:
I'm no regex-expert but I read online that this regex could cause issues for safari users. Here's the suggested solution

```suggestion
    .replace(/(?:^|-)\./, '0.') // prepend 0 when integer part is missing for decimal
```

Reply:
Fixed